File: Swift/LeetCode/LeetCode3612.swift
// https://leetcode.com/problems/process-string-with-special-operations-i

class Solution {
    func processStr(_ s: String) -> String {
        var answer = ""
        for char in s {
            switch char {
            case "*":
                guard answer.count != 0 else { break }
                answer.popLast()
            case "#":
                answer += answer
            case "%":
                answer = String(answer.reversed())
            default:
                answer.append(char)
            }
        }
        return answer
    }
}

File: cpp/leetcode/leetcode3612.cpp
// https://leetcode.com/problems/process-string-with-special-operations-i

#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
    string processStr(string s) {
        string answer = "";
        for (char c: s) {
            switch (c) {
                case '*':
                    if (answer.size() != 0) {
                        answer.pop_back();
                    }
                    break;
                case '#':
                    answer += answer;
                    break;
                case '%':
                    reverse(answer.begin(), answer.end());
                    break;
                default:
                    answer.push_back(c);
            }
        }
        return answer;
    }
};
